fotoobo/fortinet/fortinet.py

```diff
@@ -77,43 +80,49 @@ def api(  # pylint: disable=too-many-arguments
         API request to a Fortinet device.
 
         Args:
-            method:  Request method from [get, post]
-            url:     Rest API URL to request data from
-            headers: Dictionary with headers (if needed)
-            params:  Dictionary with parameters (if needed)
-            payload: JSON body for post requests (if needed)
-            timeout: The requests read timeout
+            method:     HTTP request method
+            url:        Rest API URL to request data from
+            headers:    Dictionary with headers (if needed)
+            params:     Dictionary with parameters (if needed)
+            payload:    JSON body for post requests (if needed)
+            timeout:    The requests read timeout
 
         Returns:
             Response from the request
         """
         full_url = f"{self.api_url}/{url.strip('/')}".strip("/")
-        params = params or {}
-        payload = payload or {}
         timeout = timeout or self.timeout
         start = time()
 
+        if method.upper() not in self.ALLOWED_HTTP_METHODS:
+            error = f"HTTP method '{method.lower()}' is not implemented"
```

Comment:
We usually use upper case for HTTP keywords, why not here?

Reply:
Good point.